### PR TITLE
[10.0][ADD] account_multicurrency_revaluation_rate_type

### PR DESCRIPTION
**Pull request by i-vyshnevska**
functional from https://github.com/OCA/account-closing/pull/98 was moved to separate addon 
this module allow to choose  which currency type use when revaluation i executed
also some  comments was added to account_multicurrency_revaluation

blocked by https://github.com/OCA/account-closing/pull/102

### CODE REVIEW

**Comment by i-vyshnevska** [comment]
@grindtildeath can you give some attention  to this PR

File: account_multicurrency_revaluation_type/tests/test_currency_revaluation.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        move_ids = []
+        lines = result['domain'].split(',')
+        for l in lines:
+            dig = re.search(r'\d+', l)
+            if dig:
+                move_ids.append(int(dig.group()))
+        move_ids = self.env['account.move.line'].browse(move_ids)
+        return move_ids
+
+    def wizard_execute(self, date, type):
```
Comment (sebalix) [inline account_multicurrency_revaluation_type/tests/test_currency_revaluation.py]: rename `type` (Python keyword) to `rate_type`

File: account_multicurrency_revaluation_type/tests/test_currency_revaluation.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        # 500/1.2 = 416.67 - 200 = 216.67
+
+        result = self.wizard_execute(self.apr_05, 'average')
+        move_ids = self._get_move_ids(result)
+        debit = move_ids.filtered(lambda s: s.credit > 0)
+        self.assertEqual(sum(debit.mapped('balance')), -346.67)
+
+    def _get_move_ids(self, result):
+        move_ids = []
+        lines = result['domain'].split(',')
```
Comment (sebalix) [inline account_multicurrency_revaluation_type/tests/test_currency_revaluation.py]: can't we use `safe_eval` to get the domain?

File: account_multicurrency_revaluation_type/model/res_company.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    rate_type = fields.Selection(
```
Comment (sebalix) [inline account_multicurrency_revaluation_type/model/res_company.py]: I would name this field `currency_rate_type` instead, easier to understand what we are talking about in the UI/database schema.

File: account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class WizardCurrencyRevaluationType(models.TransientModel):
+
+    _inherit = 'wizard.currency.revaluation'
+
+    # Todo get in count direct/indirect rate
```
Comment (sebalix) [inline account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py]: Is this TODO still relevant? If it's optional, can it be added to the ROADMAP section?

File: account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation_view.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_account_config_settings" model="ir.ui.view">
```
Comment (sebalix) [inline account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation_view.xml]: `view_account_currency_revaluation_wizard` (`view_account_config_settings` already used in `views/res_config_view.xml`)

**Review by sebalix — CHANGES_REQUESTED** [review]

File: account_multicurrency_revaluation_type/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    "version": "10.0.1.0.1",
+    "category": "Finance",
+    "summary": "Manage revaluation for multicurrency environment",
+    "author": "Camptocamp,Odoo Community Association (OCA)",
+    "license": 'AGPL-3',
+    "depends": [
+        "account_multicurrency_revaluation",
+        "currency_monthly_rate",
+    ],
+    "demo": [
```
Comment (mpanarin) [inline account_multicurrency_revaluation_type/__manifest__.py]: remove the empty block

File: account_multicurrency_revaluation_type/tests/test_currency_revaluation.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+from datetime import date
+import re
+from odoo.tests.common import SavepointCase
+
+
+class TestCurrencyRevaluationType(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestCurrencyRevaluationType, cls).setUpClass()
```
Comment (mpanarin) [inline account_multicurrency_revaluation_type/tests/test_currency_revaluation.py]: `cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

File: account_multicurrency_revaluation_type/tests/test_currency_revaluation.py [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+                'company_id': cls.company.id,
+            })
+            monthly_rate.create(r)
+
+        cls.jan_31 = '%s-01-31' % cls.year
+        cls.apr_05 = '%s-04-05' % cls.year
+        cls.dec_31 = '%s-12-31' % cls.year
+
+        cls.company.revaluation_loss_account_id = \
+            cls.env.ref('account_multicurrency_revaluation.acc_reval_loss').id
```
Comment (mpanarin) [inline account_multicurrency_revaluation_type/tests/test_currency_revaluation.py]: why not use the ref method you initialized earlier? )

File: account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        context = self.env.context
+
+        currency_obj = self.env['res.currency']
+        company = form.journal_id.company_id or form.env.user.company_id
+
+        # Compute unrealized gain loss
+        ctx_rate = context.copy()
+        ctx_rate['date'] = self.revaluation_date
+        ctx_rate['monthly_rate'] = True
+        # ctx_rate['revaluation'] = True
```
Comment (mpanarin) [inline account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py]: left by mistake?

**Review by mpanarin — CHANGES_REQUESTED** [review]

File: account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py (hunk 4 above)
Comment (i-vyshnevska) [reply account_multicurrency_revaluation_type/wizard/wizard_currency_revaluation.py]: yes, but it should be done in other module

**Review by Tonow-c2c — APPROVED** [review]
LGTM Just code review

File: account_multicurrency_revaluation/model/res_currency.py [hunk 10]
```diff
@@ -14,6 +14,7 @@ class ResCurrency(models.Model):
 
     @api.model
     def _get_conversion_rate(self, from_currency, to_currency):
+        # TODO get rid of this method
```
Comment (grindtildeath) [inline account_multicurrency_revaluation/model/res_currency.py]: Please go ahead. cf https://github.com/OCA/account-closing/pull/102/commits/df4e429e47f72f9df507c889d0f5b3d21e3bbca3

**Review by grindtildeath — CHANGES_REQUESTED** [review]
I guess most of my comments from https://github.com/OCA/account-closing/pull/106#pullrequestreview-221592612 also applies here.

**Comment by simahawk** [comment]
lint check :red_circle:

File: account_multicurrency_revaluation_rate_type/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Multicurrency revaluation with monthly currency rates",
+    "version": "10.0.1.0.1",
```
Comment (simahawk) [inline account_multicurrency_revaluation_rate_type/__manifest__.py:7]: if you add it now, it should be 1.0.0

File: account_multicurrency_revaluation_rate_type/model/res_company.py [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    revaluation_rate_type = fields.Selection(
+        string="Rate type", selection=[
+            ('average', 'Average'),
+            ('daily', 'Daily'),
+        ],
+        default='daily')
```
Comment (simahawk) [inline account_multicurrency_revaluation_rate_type/model/res_company.py]: same as #106

**Review by simahawk — APPROVED** [review]
skimmed thru it: LG but - as per #106 - I leave the last word to @grindtildeath :wink:

**Pull request closed without merging**